### PR TITLE
Report originally resolved module as part of reasons

### PR DESCRIPTION
**Pull request by smikula**
**What kind of change does this PR introduce?**

This addresses #8606.  Currently when modules are scope hoisted there is not enough information to reconstruct the original module dependency graph.  With this change the `reasons` structure in the reported stats contains a field for the originally resolved module.

**Did you add tests for your changes?**

Not yet, but I'm happy to if needed.  Currently not all tests seem to be passing, possible because this is the alpha branch.

**Does this PR introduce a breaking change?**

No, this should be purely additive.

**What needs to be documented once your changes are merged?**

The new property in the `reason` structure.

### CODE REVIEW

File: lib/stats/DefaultStatsFactoryPlugin.js [hunk 1]
```diff
@@ -509,6 +509,9 @@ const SIMPLE_EXTRACTORS = {
 				moduleName: reason.originModule
 					? reason.originModule.readableIdentifier(requestShortener)
 					: null,
+				resolvedModule: reason.resolvedOriginModule
```
Comment (sokra) [inline lib/stats/DefaultStatsFactoryPlugin.js]: resolvedModuleId and resolvedModuleIdentifier too

**Comment by smikula** [comment]
@sokra: I added this to the `toString` output as an optional last field in the reason line -- optional in that it only shows up if the resolved module is different from the actual module.  I've also added a test and updated the snapshots.

**Comment by smikula** [comment]
I'm not clear why the test case is failing only for Node 10 on Windows.  It looks like in that particular case the reasons are reported in a different order so they don't match the snapshot.  Is there anything that would be non-deterministic about this ordering?

File: lib/stats/DefaultStatsPrinterPlugin.js [hunk 2]
```diff
@@ -510,7 +511,8 @@ const PREFERED_ORDERS = {
 		"moduleId",
 		"module",
 		"loc",
```
Comment (sokra) [inline lib/stats/DefaultStatsPrinterPlugin.js]: Place it between `module` and `loc`

**Comment by smikula** [comment]
@sokra, thanks for the pointers.  I've made the changed you suggested and it looks like that took care of the test failure.

**Comment by smikula** [comment]
Thanks for the approval.  Is there anything else I need to do before you merge it?

**Pull request merged**